In [1]:
# Importing necessary libraries 
from pulp import *
import warnings
warnings.filterwarnings('ignore')
import pandas as pd
import matplotlib.pyplot as plt
import pulp
import os

In [3]:
x1 = pulp.LpVariable('x1', lowBound = 0, upBound = 8) #No. of hours the old facility is operated 
x2 = pulp.LpVariable('x2', lowBound = 0) #No. of hours the new facility is operated
x3 = pulp.LpVariable('x3', lowBound = 0)  #No. of hours the single purpose facility is operated
cost = pulp.LpProblem("Minimum Cost", pulp.LpMinimize)
#Minimizing cost
cost += 190*x1 + 260*x2 + 150*x3, 'Objective Function'
cost += 295*x1 + 385*x2 + 350*x3 >=5500, 'Amount of powder mix produced' # as per the ordered amount of powder mix
cost += 260*x1 + 375*x2 >= 4000, "Amount of tea concentrate produced" # as per the ordered amount of tea concentrate 
cost.solve()
print('Status',LpStatus[cost.status])
# to check values of decision variables and minimum production cost 
for i in cost.variables(): 
    print(i.name, '=', i.varValue)
print('Objective Function = ', value(cost.objective))

Status Optimal
x1 = 8.0
x2 = 5.12
x3 = 3.3394286
Objective Function =  3352.11429


b) Values of the decision variables are x1(old facility) = 8 hours, x2(New facility) = 5.12 hours, x3(single purpose facility) = 3.33 hours. The minimum production cost is $ 3352.11429

In [4]:
l = list()
for name,c in cost.constraints.items():
    l.append([name, str(cost.constraints[name]), c.pi, c.slack])
pd.DataFrame(l, columns = ['name', 'constraints', 'shadow_price','slack'])

name                       constraints  \
0       Amount_of_powder_mix_produced  295*x1 + 385*x2 + 350*x3 >= 5500   
1  Amount_of_tea_concentrate_produced           260*x1 + 375*x2 >= 4000   

   shadow_price  slack  
0      0.428571   -0.0  
1      0.253333   -0.0